src/rule_dsl.jl

```diff
@@ -248,40 +248,86 @@ struct RuleRewriteError
     expr
 end
 
-function (r::RuleSet)(term, context=EmptyCtx(); depth=typemax(Int), applyall=false, recurse=true)
+pick(a, b) = a
+pick(a::Nothing, b) = b
```

Comment:
I think using dispatch for this is a bad idea. It's always a dynamic dispatch if the nothing comes from `(::RuleSet)(term)` 
```julia
pick(@nospecialize(a), b) = a === nothing ? b : a
```

Reply:
Maybe small union splitting helps here but I don't know.

Reply:
gonna have to benchmark that.